browser/blueprints.py

```diff
@@ -294,46 +272,58 @@ def shortcut(filename):
     request to access a specific data file that has been preloaded to the
     input S3 bucket (ex. http://127.0.0.1:5000/test.npz).
     """
-    rgb = request.args.get('rgb', default='false', type=str)
-    pixel_only = request.args.get('pixel_only', default='false', type=str)
-    label_only = request.args.get('label_only', default='false', type=str)
-
-    settings = {
-        'rgb': bool(distutils.util.strtobool(rgb)),
-        'pixel_only': bool(distutils.util.strtobool(pixel_only)),
-        'label_only': bool(distutils.util.strtobool(label_only))
-    }
-
-    if is_trk_file(filename):
-        filetype = 'track'
-        title = 'Tracking Tool'
-
-    elif is_npz_file(filename):
-        filetype = 'zstack'
-        title = 'Z-Stack Tool'
-
-    else:
-        # TODO: render an error template instead of JSON.
-        error = {
-            'error': 'invalid file extension: {}'.format(
-                os.path.splitext(filename)[-1])
-        }
-        return jsonify(error), 400
-
+    settings = make_settings(filename)
     return render_template(
         'tool.html',
-        filetype=filetype,
-        title=title,
-        filename=filename,
         settings=settings)
 
 
 def get_edit(project):
     """Factory for Edit objects"""
-    filename = project.filename
+    filename = project.path
     if is_npz_file(filename):
         return ZStackEdit(project)
     elif is_trk_file(filename):
         # don't use RGB mode with track files
         return TrackEdit(project)
     return BaseEdit(project)
+
+
+def make_settings(filename):
+    """Returns a dictionary of settings to send to the front-end."""
+    folders = re.split('__', filename)
+
+    input_bucket = folders[0] if len(folders) > 1 else S3_INPUT_BUCKET
```

Comment:
Most files are stored in subfolders, so I think this logic for the bucket names could create confusion. Maybe we could use `S3_INPUT_BUCKET` if `folders[0] == ''`? The urls to create those paths would not look pretty but it would avoid bucket/subfolder ambiguity

Reply:
I agree that this is a little unclear, but as mentioned below the URLs/sub folders will probably not be an issue until the file loader is merged in. Maybe we could just add a TODO here to take care of it in the next PR for file uploading?

Reply:
Yep, fine by me